### PR TITLE
Feature/add action for retrieving table data

### PR DESCRIPTION
**Pull request by StasKozar**
https://trello.com/c/f9VWMjCQ/210-add-action-for-retrieving-tabledata

### CODE REVIEW

**Comment by TatsuUkraine** [comment]
It seems you branch is extended from https://github.com/BinaryStudioAcademy/bsa-2019-metrica/pull/206 which means that you PR contains exact same issues, don't do this anymore, since you PR will be blocked until #206 will be merged

File: frontend/src/store/modules/visitors/actions.js [hunk 1]
```diff
@@ -16,4 +33,30 @@ export default {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [FETCH_TABLE_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+        context.commit(SET_TABLE_DATA_FETCHING);
+        context.commit(GET_ACTIVE_BUTTON)
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: you will not get data wit this approach. Take active button from context.state

File: frontend/src/store/modules/visitors/actions.js [hunk 2]
```diff
@@ -16,4 +33,30 @@ export default {
...
+            })
+            .then(response => {
+                context.commit(SET_TABLE_DATA, response.data);
+            })
+            .catch(err => {
+                context.commit(RESET_TABLE_DATA_FETCHING);
+                throw err;
+            })
+            .finally(() => {
+                context.commit(RESET_TABLE_DATA_FETCHING);
```
Comment (lenchvolodymyr) [inline frontend/src/store/modules/visitors/actions.js]: probably better to move it in then, as if error will occur the error can be lost and unhandled

File: frontend/src/store/modules/visitors/actions.js (hunk 1 above)
Comment (StasKozar) [reply frontend/src/store/modules/visitors/actions.js]: Fixed

File: frontend/src/store/modules/visitors/actions.js (hunk 2 above)
Comment (StasKozar) [reply frontend/src/store/modules/visitors/actions.js]: Moved

**Pull request merged**